packages/hub/db/migrations/20220728144935996_add-profiles.ts

```diff
@@ -0,0 +1,81 @@
+import { MigrationBuilder, ColumnDefinitions } from 'node-pg-migrate';
+
+export const shorthands: ColumnDefinitions | undefined = undefined;
+
+const idField = { id: { type: 'uuid', primaryKey: true } };
+
+const merchantInfosFields = {
+  name: { type: 'string', notNull: true },
+  slug: { type: 'string', notNull: true },
+  color: { type: 'string', notNull: true },
+  text_color: { type: 'string', notNull: true },
+  owner_address: { type: 'string', notNull: true },
+};
+
+const cardSpaceFields = {
+  links: { type: 'json[]', notNull: true, default: '{}' },
+  profile_image_url: { type: 'string' },
+  profile_description: { type: 'string' },
+};
+
+export async function up(pgm: MigrationBuilder, run: () => Promise<void>): Promise<void> {
+  const createdAtField = { created_at: { type: 'timestamp', notNull: true, default: pgm.func('current_timestamp') } };
+
+  pgm.createTable('profiles', {
+    ...idField,
+    ...merchantInfosFields,
+    ...cardSpaceFields,
+    ...createdAtField,
+  });
+
+  await forceRun(pgm, run);
+
+  await pgm.db.query(`
+    INSERT INTO profiles
+    SELECT
+      m.id, m.name, m.slug, m.color, m.text_color, m.owner_address,
+      c.links, c.profile_image_url, c.profile_description,
+      m.created_at
+    FROM merchant_infos as m
+    LEFT JOIN card_spaces AS c ON m.id = c.merchant_id
+  `);
+
+  await pgm.db.query('DROP TABLE card_spaces');
```

Comment:
I'm extra paranoid about dropping tables in production before verifying the migration + new code works ok 🫥

Reply:
yes, it’s scary! I ran this many times locally to refine it. I’ll be careful about preserving everything before running it in production.